controllers/space/space_controller.go

```diff
@@ -129,13 +129,37 @@ func (r *Reconciler) addFinalizer(logger logr.Logger, space *toolchainv1alpha1.S
 // - the Space's `Ready=false/Updating` condition is too recent, so we're not sure that the NSTemplateSetController was triggered.
 // Returns `false` otherwise
 func (r *Reconciler) ensureNSTemplateSet(logger logr.Logger, space *toolchainv1alpha1.Space) (bool, error) {
+	// deprovision from space.Status.TargetCluster if needed
+	if space.Status.TargetCluster != "" && space.Spec.TargetCluster != space.Status.TargetCluster {
+		logger.Info("retargeting space", "from_cluster", space.Status.TargetCluster, "to_cluster", space.Spec.TargetCluster)
+		// look-up and delete the NSTemplateSet on the current member cluster
+		if isBeingDeleted, err := r.deleteNSTemplateSetFromCluster(logger, space, space.Status.TargetCluster); err != nil {
+			return false, r.setStatusRetargetFailed(logger, space, err)
+		} else if isBeingDeleted {
+			logger.Info("wait while NSTemplateSet is being deleted", "member_cluster", space.Status.TargetCluster)
+			return false, r.setStatusRetargeting(space)
+		} else {
```

Comment:
It's just a detail, but this last `else {` is not needed, because you always return in the previous cases.

Reply:
right, but I like the structure of the `if/else if/else`, so I may keep it as it is, unless it's really an issue.